### ISSUE TITLE
Encoder code incorrect?

### ISSUE THREAD

**Issue by MartinRusk** [brunomlopes/pro-micro-surface-dialish#1]
Hi. Upon reviewing your Code I think in line 60 there might be missing the clipping to the lower 4 bits:
int sum = ((previous << 2) | encoded) & 0x0f;

**Comment by brunomlopes** [comment]
Hi, I might be? I haven't used this in a while, and the code was taken from the sample.

**Comment by MartinRusk** [comment]
Yeah, I found the same bug in the sample code ... btw, I have just published my Surface Dial design on printables: https://www.printables.com/de/model/371867-surface-dial-alternative
